### PR TITLE
Major updates to code

### PR DESCRIPTION
**Pull request by rbedfordpro**
Major updates to code:
Added more vendors / stable master
Added more portals
Added automation post-purchase of guild house: new house now starts with 1 portal & guild assistant
Added ability to allow other guild ranks to make purchases via config
Guild House now deletes objects & creatures upon selling guild house or guild disband
Updated Gossip code

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
I'm not commenting on the quality of the code but next time, try to make different PRs for different things

Like Travis in one part, feature in another, fix in another etc

Though for modules it's less important, just saying :)

**Comment by FrancescoBorzi** [comment]
I agree with @BarbzYHOOL . Smaller PRs are easier to be reviewed, tested, etc.. so they can be merged faster.

**Comment by rbedfordpro** [comment]
I'll definitely remember that for the future. In this case though it would have been about 15+ PRs with all the changes WiZZy and I did in revamping so much of it and adding what was put in.

**Comment by Stoabrogga** [comment]
I'll review this PR when I have time.
Btw, why are existing creature templates used (18649,33251)? Why not create new ones? I think it would be better to not manipulate the data of the core, but use separate entries for the module in order to easily revert the changes if the module is not used anymore.

**Comment by milestorme** [comment]
tested and good , thanks @rbedfordpro

File: sql/world/guild_house_spawns.sql [hunk 1]
```diff
@@ -44,7 +44,7 @@ CREATE TABLE IF NOT EXISTS `guild_house_spawns` (
   UNIQUE KEY `entry` (`entry`)
 ) ENGINE=InnoDB AUTO_INCREMENT=51 DEFAULT CHARSET=utf8;
 
--- Dumping data for table acore_world.guild_house_spawns: ~29 rows (approximately)
+-- Dumping data for table acore_world.guild_house_spawns: ~56 rows (approximately)
```
Comment (BarbzYHOOL) [inline sql/world/guild_house_spawns.sql]:
```suggestion
-- Dumping data for table acore_world.guild_house_spawns: ~56 rows (approximately)
```

**Comment by BarbzYHOOL** [comment]
Yes, these two creatures are existing 18649,33251 but it's not rbedfordpro who added them, it's the original author's idea.

Monica apparently is supposed to do an emote during a fight in old hillsbrad (lol)
The DK trainer is for the tournament server (so it could serve a purpose too)

So yes, they are almost useless ingame but it might be better to use other creatures

As for the rest of the PR, I checked quickly and it looks good, I still haven't tested yet though

**Comment by BarbzYHOOL** [comment]
I think this can be merged, i didn't test in details but i tested quickly all the features, it works quite well (i didn't try all vendors etc though)

**Comment by BarbzYHOOL** [comment]
After anyone answers here, i'll merge this :P

**Pull request merged**